server/plugin/api.go

```diff
@@ -1218,12 +1280,21 @@ func (p *Plugin) getMilestones(c *UserContext, w http.ResponseWriter, r *http.Re
 	p.writeJSON(w, allMilestones)
 }
 
-func getRepositoryList(c context.Context, userName string, githubClient *github.Client, opt github.ListOptions) ([]*github.Repository, error) {
+func (p *Plugin) getRepositoryList(c context.Context, ghInfo *GitHubUserInfo, userName string, githubClient *github.Client, opt github.ListOptions) ([]*github.Repository, error) {
 	var allRepos []*github.Repository
 	for {
-		repos, resp, err := githubClient.Repositories.List(c, userName, &github.RepositoryListOptions{ListOptions: opt})
-		if err != nil {
-			return nil, err
+		var repos []*github.Repository
+		var resp *github.Response
+		var err error
+		cErr := p.useGitHubClient(ghInfo, func(info *GitHubUserInfo, token *oauth2.Token) error {
+			repos, resp, err = githubClient.Repositories.List(c, userName, &github.RepositoryListOptions{ListOptions: opt})
+			if err != nil {
+				return err
+			}
+			return nil
+		})
+		if cErr != nil {
+			return nil, cErr
```

Comment:
Do we need to log the error here or it is handled somewhere else?

Reply:
it is logged from where it is called